testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -177,3 +177,20 @@ def test_presence_griddata(self):
             except ImportError:
                 self.fail('''MDAnalysis.analysis.density should not raise
                              an ImportError if gridData is available.''')
+
+#issue 1231
+def test_check_set_unit_keyerror():
+    import MDAnalysis.analysis.density
+    D = MDAnalysis.analysis.density.Density(np.zeros((2,2)), np.zeros((2,2)),
+                                                    parameters={'isDensity': False},
```

Comment:
Would the test be valid id you do not provide `parameters` and `unit` at the instance creation? Then you would call assert error with the unit dictionary directly rather than with `D.unit`.